src/components/internal/ImageUploader.component.tsx

```diff
@@ -0,0 +1,163 @@
+    fileUploadSessionFactory: FileUploadSessionFactory;
+    uploadLabel: string;
+    onFileUploadInit: () => Promise<number>;
+    onFileUploadFinished: (fileUploadSessionId: number) => Promise<void>;
+    imageValidator?: (image: HTMLImageElement) => boolean;
+    imageAlt?: string;
+    imageUrl?: string;
+    placeholder?: string;
+}
+
+interface State {
+    isPreviewingImage: boolean;
+    previewImageUrl?: string;
+    uploadProgress: Progress;
+}
+
+export class ImageUploaderComponent extends Component<Props, State> {
+    private readonly fileUploadSessionFactory: FileUploadSessionFactory;
+
+    constructor(props: Props) {
+        super(props);
+        this.fileUploadSessionFactory = props.fileUploadSessionFactory;
+        this.state = {
+            isPreviewingImage: false,
+            uploadProgress: {
+                newBytesUploaded: 0,
+                totalPercentage: 0,
+            },
+        };
+    }
+
+    private renderPreview() {
+        return (
+            <Fragment>
+                <img
+                    alt={this.props.imageAlt}
+                    className={styles.ActiveImg}
+                    src={this.state.previewImageUrl}
+                />
+                <div
+                    className={styles.UploadProgress}
+                    style={{
+                        top: `${this.state.uploadProgress.totalPercentage}%`,
+                    }}
+                />
+            </Fragment>
+        );
+    }
+
+    private renderActiveImage() {
+        return (
+            <Fragment>
+                {this.props.imageUrl ? (
+                    <img
+                        alt={this.props.imageAlt}
+                        className={styles.ActiveImg}
+                        src={this.props.imageUrl}
+                    />
+                ) : (
+                    <div className={styles.Placeholder}>
+                        {this.props.placeholder || ""}
+                    </div>
+                )}
+                <div className={styles.UploadButton}>
+                    <input
+                        type={'file'}
+                        className={styles.FileInput}
+                        title=""
+                        accept="image/png,image/jpeg,image/bmp,image/svg+xml"
+                        onChange={this.onProfileImageChange}
+                    />
+                    <div className={styles.Label}>{this.props.uploadLabel}</div>
+                </div>
+            </Fragment>
+        );
+    }
+
+    private onProfileImageChange = async (
+        event: ChangeEvent<HTMLInputElement>,
+    ) => {
+        const files = event.target.files;
+        if (!files || files.length < 1) {
+            return;
+        }
+
+        const file = files[0];
+        const previewImageUrl = URL.createObjectURL(file);
+        const image = await loadImage(previewImageUrl);
+
+        if (this.props.imageValidator && !this.props.imageValidator.call(this, image)) {
```

Comment:
```suggestion
        if (this.props.imageValidator && !this.props.imageValidator.call(null, image)) {
```

Let's pass null instead. External parent shouldn't have access to the data inside component.